# Exercise 2.2

The following are the proportions of girl births in Vienna for each month in 1908 and 1909 (out of an average of 3900 births per month):

.4777 .4875 .4859 .4754 .4874 .4864 .4813 .4787 .4895 .4797 .4876 .4859 .4857 .4907 .5010 .4903 .4860 .4911 .4871 .4725 .4822 .4870 .4823 .4973

The data are in the folder girls. von Mises (1957) used these proportions to claim that the sex ratios were less variable than would be expected by chance.

(a) Compute the standard deviation of these proportions and compare to the standard deviation that would be expected if the sexes of babies were inde- pendently decided with a constant probability over the 24-month period.

(b) The actual and theoretical standard deviations from (a) differ, of course. Is this difference statistically significant? (Hint: under the randomness model, the actual variance should have a distribution with expected value equal to the theoretical variance, and proportional to a χ2 with 23 degrees of freedom.)

In [17]:
import numpy as np

girl_births_observed = [.4777, .4875, .4859, .4754, .4874, .4864,
               .4813, .4787, .4895, .4797, .4876, .4859, 
               .4857, .4907, .5010, .4903, .4860, .4911, 
               .4871, .4725, .4822, .4870, .4823, .4973]

monthly_average_births = 3900

In [51]:
# Part A
girl_births_obs_std_dev = np.std(girl_births_observed)

girl_births_mean = np.mean(girl_births_observed)

# The variance of a sample mean for a binomial distribution is kqp/n.
# For a bernoulli trial, k = 1
# The standard error is the square root of this.

girl_births_std_error_expected = np.sqrt(girl_births_mean*(1-girl_births_mean)/monthly_average_births)

The expected deviation if the sexes of the babies were independently decided with constant probability is 0.008. The observed deviation is 0.006.

In [61]:
# Part B
alpha = 0.05
upper_bound = stats.chisquare.
lower_bound = stats.chisquare(1 - (alpha/2), dof)

In [79]:
from scipy import stats

dof = len(girl_births_observed) - 1

upper_stat = stats.chi2.ppf(alpha/2, dof)
lower_stat = stats.chi2.ppf(1-(alpha/2), dof)
test_statistic = dof*np.power((girl_births_obs_std_dev/girl_births_std_error_expected),2)

print(upper_stat)
print(lower_stat)
print(test_statistic)

11.6885519225
38.0756272504
14.138532976


Because the test statistic (14.139) is not lower than 11.689 nor higher than 38.07, we do not reject the null hypothesis.